src/Components/ProductTile.js

```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
 //          - NA
 const ProductTile = ({ products, index, setProducts }) => {
   const history = useHistory();
-  const upVote = () => {
-    const updatedProduct = { ...products[index] };
-    let currentVote = updatedProduct.upVoted ? 1 : (updatedProduct.downVoted ? -1 : 0);
-    updatedProduct.upVoted = !updatedProduct.upVoted;
-    updatedProduct.downVoted = false;
-    let newVote = updatedProduct.upVoted ? 1 : (updatedProduct.downVoted ? -1 : 0);
-    updatedProduct.votes = updatedProduct.votes - currentVote + newVote;
-    setProducts(products.map((product) => product.id === products[index].id ? updatedProduct : product));
+
+  const username = ReactSession.get("username");
+  const loggedin = username !== ''?true:false;
+  const [open, setOpen] = React.useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
   };
-  const downVote = () => {
-    const updatedProduct = { ...products[index] };
-    let currentVote = updatedProduct.upVoted ? 1 : (updatedProduct.downVoted ? -1 : 0);
-    updatedProduct.downVoted = !updatedProduct.downVoted;
-    updatedProduct.upVoted = false;
-    let newVote = updatedProduct.upVoted ? 1 : (updatedProduct.downVoted ? -1 : 0);
-    updatedProduct.votes = updatedProduct.votes - currentVote + newVote;
-    setProducts(products.map((product) => product.id === products[index].id ? updatedProduct : product));
+
+  const handleClose = () => {
+    setOpen(false);
   };
+
+  const increaseUpVote = () =>{
+    const form = new FormData();
+    console.log(products[index].uid);
+    form.append("uid", products[index].uid);
+    Service.post('addTotalVote', form)
+      .then(data => 
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
      .then((data) => 
```